script/manifest/manifestmodifier.js

```diff
@@ -0,0 +1,59 @@
+define('bigscreenplayer/manifest/manifestmodifier',
+  function () {
+    'use strict';
+
+    function filter (manifest, representationOptions) {
+      var constantFps = representationOptions.constantFps;
+      var maxFps = representationOptions.maxFps;
+
+      if (constantFps || maxFps) {
+        manifest.Period.AdaptationSet = manifest.Period.AdaptationSet.map(function (adaptationSet) {
+          if (adaptationSet.contentType === 'video') {
+            var frameRates = [];
+
+            adaptationSet.Representation_asArray = adaptationSet.Representation_asArray.filter(function (representation) {
+              if (!maxFps || representation.frameRate <= maxFps) {
+                frameRates.push(representation.frameRate);
+                return true;
+              }
+            }).filter(function (representation) {
+              return !constantFps || representation.frameRate === Math.max.apply(null, frameRates);
+            });
+          }
+          return adaptationSet;
+        });
+      }
+      return manifest;
+    }
+
+    function extractBaseUrl (manifest) {
+      return manifest.Period && manifest.Period.BaseURL || manifest.BaseURL && manifest.BaseURL.__text;
+    }
+
+    function generateBaseUrls (manifest, sources) {
+      var baseUrl = extractBaseUrl(manifest);
+      if (!baseUrl) return;
+      if (baseUrl.match(/^https?:\/\//)) {
+        sources = sources.slice(0, 1);
```

Comment:
We need to modify the manifest to have the correct `serviceLocation` even though the url is not changing. This is so we know which events to ignore in `msestrategy -> propagateCdnFailover`